src/Function/Definitions.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions for types of functions
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary
+
+module Function.Definitions
+  {a b ℓ₁ ℓ₂} {A : Set a} {B : Set b} (_≈₁_ : Rel A ℓ₁) (_≈₂_ : Rel B ℓ₂)
+  where
+
+open import Data.Product using (∃; _×_)
+open import Level using (_⊔_)
+
+------------------------------------------------------------------------
+-- Definitions
+
+Congruent : (A → B) → Set (a ⊔ ℓ₁ ⊔ ℓ₂)
+Congruent f = ∀ {x y} → x ≈₁ y → f x ≈₂ f y
+
+Injective : (A → B) → Set (a ⊔ ℓ₁ ⊔ ℓ₂)
+Injective f = ∀ {x y} → f x ≈₂ f y → x ≈₁ y
+
+Surjective : (A → B) → Set (a ⊔ b ⊔ ℓ₂)
+Surjective f = ∀ x → ∃ λ y → y ≈₂ f x
```

Comment:
`Surjective`, `LeftInverses` and `RightInverses` don't actually depend on both equalities. However if you move the equalities down to the definition level then it's not possible to open the module with parameters which I imagine will be quite useful...

Reply:
Yup, sounds good!

Reply:
Split them out in their own core modules.